include/rocksdb/convenience.h

```diff
@@ -27,6 +30,15 @@ struct Options;
 // of the serialization (e.g. delimiter), and how to compare
 // options (sanity_level).
 struct ConfigOptions {
+  // Constructs a new ConfigOptions with a new object registry.
+  // This method should only be used when a DBOptions is not available,
+  // else registry settings may be lost
+  ConfigOptions();
```

Comment:
Also indicate in the ConfigOptions(const DBOptions&) comment, a new object registry is also created.